receiver/mongodbatlasreceiver/receiver.go

```diff
@@ -78,46 +82,100 @@ func (s *mongodbatlasreceiver) shutdown(context.Context) error {
 }
 
 func (s *mongodbatlasreceiver) poll(ctx context.Context, time timeconstraints) error {
-	orgs, err := s.client.Organizations(ctx)
-	if err != nil {
-		return fmt.Errorf("error retrieving organizations: %w", err)
-	}
-	for _, org := range orgs {
-		projects, err := s.client.Projects(ctx, org.ID)
-		if err != nil {
-			return fmt.Errorf("error retrieving projects: %w", err)
-		}
-		for _, project := range projects {
-			nodeClusterMap, providerMap, err := s.getNodeClusterNameMap(ctx, project.ID)
+	if len(s.cfg.Projects) > 0 {
```

Comment:
Rather than if / else both wrapping large blocks of code, we should only need an `if` which returns at the end.

Similarly, these are complex enough that it would be helpful to move them into separate functions:

```
func (s *mongodbatlasreceiver) poll(ctx context.Context, time timeconstraints) error {
	if len(s.cfg.Projects) == 0 {
		return s.pollAllProjects(ctx, time)
	}
	return s.pollProjects(ctx, time)
}
```

Reply:
updated code where I could to make it look better.